### PR TITLE
Misc. Logging Changes

### PR DESCRIPTION
**Pull request by iguessthislldo**
- Log when Readers and Writers get a GUID. Also logs the topic type and topic name.
- Fix missing dash before metadata in OpenDDS version logging.
- Make all thread status logging I could find start at 4 to match the update logging point.
- Slap `DCPS_debug_level` on some logging points that were missing them.

### CODE REVIEW

**Review by mitza-oci — APPROVED** [review]

**Comment by mitza-oci** [comment]
Are the errors detected in `tests/security/attributes` (GHPRB) new from this PR?

**Comment by iguessthislldo** [comment]
I think they're expected to fail, but I changed it from a warning to an error to match the other error returns that log in that function.

**Comment by iguessthislldo** [comment]
I will look into more and fix it.

File: dds/DCPS/DataWriterImpl.cpp [hunk 1]
```diff
@@ -1482,16 +1485,22 @@ DataWriterImpl::enable()
...
+    }
     data_container_->shutdown_ = true;
     return DDS::RETCODE_ERROR;
   }
 
+  if (DCPS_debug_level >= 2) {
+    ACE_DEBUG((LM_DEBUG, "(%P|%t) DataWriterImpl::enable: "
+      "got GUID %C, publishing to topic name \"%C\" type \"%C\"\n",
+      LogGuid(publication_id_).c_str(),
+      topic_servant_->get_name(), topic_servant_->get_type_name()));
```
Comment (mitza-oci) [inline dds/DCPS/DataWriterImpl.cpp:1501]: Use `_var` or the methods that don't start with `get_*` from `TopicImpl.h`

**Review by mitza-oci — CHANGES_REQUESTED** [review]

**Review by mitza-oci — APPROVED** [review]

**Pull request merged**